src/test/java/org/cactoos/scalar/TernaryTest.java

```diff
@@ -73,14 +76,40 @@ void conditionBoolean() throws Exception {
 
     @Test
     void conditionFunc() throws Exception {
-        MatcherAssert.assertThat(
+        new Assertion<>(
+            "Must call the Func with the input",
             new Ternary<>(
                 5,
                 input -> input > 3,
-                input -> input = 8,
-                input -> input = 2
-            ).value(),
-            Matchers.equalTo(8)
-        );
+                input -> input + 1,
+                input -> input + 2
+            ),
+            new ScalarHasValue<>(6)
+        ).affirm();
+    }
+
+    @Test
+    void conditionSticky() throws Exception {
+        final AtomicInteger counter = new AtomicInteger(0);
+        new Assertion<>(
+            "Must conserve the same value for the whole evaluation",
+            new Ternary<String>(
+                counter::incrementAndGet,
+                i -> i == 1,
+                i ->  i + " equals " + 1,
+                i -> "else: " + i
+            ),
+            new ScalarHasValue<>("1 equals 1")
+        ).affirm();
+        new Assertion<>(
+            "Must conserve the same new value for the whole evaluation",
```

Comment:
@victornoel It's not clear here what should be conserved. Shouldn't `Ternary` be assigned to a variable for `Sticky` to have any effect?

Reply:
@andreoss ok, I see it's not clear. I thought the purpose of the PR was clear based on the ticket description, but maybe it's only clear in my head ^^ I will try to explain here and then once it's clear also for you, I will imrpove the code.

Basically, when you can `new Ternary(xxx).value()`, you want that for every branches of the ternary, the value used to be the same. So for example in the test, each call to the provided scalar.value will return a new value, but you want that each call to ternary.value gives you the result for one given value of the provided scalar.value. Like this it makes behaviour much closer to the expected one by someone discovering the class ternary.

Also it allows to express things that were not possible before, we will see that in the PR for #1460: it will allow to write it only by pure object composition without imperative intermediate values:
```java
     public Abbreviated(final Text text, final int max) {
         super(
             new Flatten(
                new Ternary<Text>(
                    () -> new Sticky(text),
                    t -> t.asString().length() <= max,
                    t -> t,
                    t -> new FormattedText(
                        "%s%s",
                        new Sub(
                            t,
                            0,
                            max - Abbreviated.ELLIPSES.length()
                        ),
                        Abbreviated.ELLIPSES
                    )
                )
             )
        )
    }
```

Reply:
@andreoss actually, thank you for your comment because I realised afterward that it's not that it was clear, but as you said, that the test was incorrect: ternary should be assigned to a variable to be reused in the two assertions. And when I tried, it showed that my implementation was indeed incorrect :)

I've fixed that in my last push!